test/constraint/encoding.test.ts

```diff
@@ -444,6 +444,201 @@ describe('constraints/encoding', () => {
     });
   });
 
+  describe('scalePropertiesSupportedByScaleType', () => {
+    describe('scaleBandSize', () => {
+      let encQ: EncodingQuery = {
+        channel: Channel.X,
+        field: 'A',
+        type: Type.QUANTITATIVE,
+        scale: {bandSize: 10, type: undefined}
+      };
+
+      it('should return false if scaleType does not suport scaleBandSize', () => {
+        [ScaleType.LINEAR, ScaleType.LOG, ScaleType.POW, ScaleType.QUANTILE, ScaleType.QUANTIZE,
+         ScaleType.SQRT, ScaleType.TIME, ScaleType.UTC]. forEach((scaleType) => {
+          (encQ.scale as ScaleQuery).type = scaleType;
+          assert.isFalse(ENCODING_CONSTRAINT_INDEX['scalePropertiesSupportedByScaleType'].satisfy(encQ, schema, defaultOpt));
+        });
+      });
+
+      it('should return true if scaleType supports scaleBandSize', () => {
+        (encQ.scale as ScaleQuery).type = ScaleType.ORDINAL;
+        assert.isTrue(ENCODING_CONSTRAINT_INDEX['scalePropertiesSupportedByScaleType'].satisfy(encQ, schema, defaultOpt));
+      });
+    });
+
+    describe('scaleClamp', () => {
+      let encQ: EncodingQuery = {
+        channel: Channel.X,
+        field: 'A',
+        type: Type.QUANTITATIVE,
+        scale: {clamp: true, type: undefined}
```

Comment:
You're setting type afterward anyway, I'm not sure why you set `type=undefined` here?   (Same for others.)